### ISSUE TITLE
Access control: `read` at field level not working properly

### ISSUE THREAD

**Issue by smoglica** [payloadcms/payload#1369]
# Bug Report

I would like to not see a field if it's me or vice versa.

## Current Behavior

I would like to see the field only if it's not me with the following rule:

```ts
access: {
    read: ({ req, id }) => {
        return req.user?.id !== id;
    },
}
```

I see the field even if it's me.

## Expected Behavior

I would expect to **not** see the field if it's me.

## Possible Solution

I've just notice that `access.read` method it's invoked 3 times. The first time `id` it's `undefined`

## Steps to Reproduce

1. Create a project with `npx create-payload-app` and choose as project template `blog`
2. Then update the `Users` collection and provide an `access.read` at `name` field level

```ts
import { CollectionConfig } from 'payload/types';

const Users: CollectionConfig = {
  slug: 'users',
  auth: true,
  admin: {
    useAsTitle: 'email',
  },
  access: {
    read: () => true,
  },
  fields: [
    // Email added by default
    {
      name: 'name',
      type: 'text',
      access: {
        read: ({ req, id }) => {
          return req.user?.id !== id;
        },
      },
    }
  ],
};

export default Users;
```

## Detailed Description

- Using `payload` version `1.1.21`

**Comment by jmikrut** [comment]
Hey @smoglica — this is actually currently expected, although we are working to improve this behavior.

See [this issue](https://github.com/payloadcms/payload/issues/575) for more info.  I just opened a `planned` GitHub discussion to go into detail about how we will rectify this into the future.

https://github.com/payloadcms/payload/discussions/1379

TL;DR: our responsive admin UI only depends on access control that is based off user roles, not data on the document currently being edited (in your case, the `id`). 

We can improve this, but as of now, this is less an issue and more a feature request - so I'm going to close this issue and keep the discussion going on the `planned` Discussion. 

We'll tackle this ASAP though as I agree it's important and would be great to have!

**Comment by JarrodMFlesch** [comment]
Hey @smoglica, 1.3.2 just released, and this functionality is now available. Document level access now runs when a doc is loaded and after it is saved! Field level access control is now document aware so you are able to exactly what you are requesting above 👍

**Comment by smoglica** [comment]
@JarrodMFlesch 

Nice! Thank you :)
